ci/buildspec_main.yml

```diff
@@ -0,0 +1,53 @@
+# AWS CodeBuild config
+# This file is referenced in the infra repo as a CodeBuild buildspec target.
+
+# These 5 variables are set in the codebuild project definition's environment secion.
+# ECR_REGION
+# ECR_ACCOUNT_ID
+# ECR_NAMESPACE
+# CLOUDINARY_NAME
+# FILE_SERVICE
+
+version: 0.2
+env:
+  shell: bash
+phases:
+  install:
+    runtime-versions:
+      nodejs: 16
+  pre_build:
+    commands:
+      - aws ecr get-login-password --region "${ECR_REGION}" | docker login --username AWS --password-stdin "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com"
+
+  build:
+    commands:
+      # Backend
+      - docker build backend/core -f build/docker/Dockerfile.backend --target test -t backend/core:test
+      - docker run backend/core:test yarn test
+      - docker build backend/core -f build/docker/Dockerfile.backend --target run -t backend/core:run
+      - docker tag backend/core:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
+      - docker tag backend/core:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:run"
+
+      # # Public
+      - docker build . -f build/docker/Dockerfile.sites-generic --target test --build-arg SITE=public --build-arg CLOUDINARY_NAME=${CLOUDINARY_NAME} -t sites/public:test
+      - docker run sites/public:test
+      - docker build . -f build/docker/Dockerfile.sites-generic --target run --build-arg SITE=public --build-arg CLOUDINARY_NAME=${CLOUDINARY_NAME} --build-arg FILE_SERVICE=${FILE_SERVICE} -t sites/public:run
+      - docker tag sites/public:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/public:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
+      - docker tag sites/public:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/public:run"
+
+      # # Partners
+      - docker build . -f build/docker/Dockerfile.sites-generic --target test --build-arg SITE=partners --build-arg CLOUDINARY_NAME=${CLOUDINARY_NAME} -t sites/partners:test
+      - docker run sites/partners:test
+      - docker build . -f build/docker/Dockerfile.sites-generic --target run --build-arg SITE=partners --build-arg CLOUDINARY_NAME=${CLOUDINARY_NAME} --build-arg FILE_SERVICE=${FILE_SERVICE} -t sites/partners:run
+      - docker tag sites/partners:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/partners:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
+      - docker tag sites/partners:run "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/partners:run"
+
+      # Pushing images to ECR
+      - docker push "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:${CODEBUILD_RESOLVED_SOURCE_VERSION:0:8}"
+      - docker push "${ECR_ACCOUNT_ID}.dkr.ecr.${ECR_REGION}.amazonaws.com/${ECR_NAMESPACE}/backend:run"
```

Comment:
We had talked about updating the tag just prior to deploy to prevent inconsistent states, but I'm fine leaving that for future updates.